Boost 1.70 support

JonasProgrammer opened the pull request:
Fixups for Boost 1.70 (which are downwards compatible to boost 1.67):
* updated `resourece_pool` (likely has to be re-based back to the ancestor repository)
* use `get_executor().context()` instead of deprecated/removed `get_io_context()`
* fix `test_asio.h` to always have an executor to make io context retrieval work

Fixes #134

thed636 commented on include/ozo/connection.h:
```diff
@@ -433,7 +433,11 @@ inline decltype(auto) get_socket(T&& conn) noexcept {
     return get_socket(conn).get_io_context();
+#else
+    return get_socket(conn).get_executor().context();
+#endif
```
Looks like this condition is not needed, because in Boost 1.66  `boost::asioposix::stream_descriptor::get_executor()` [already exists](https://www.boost.org/doc/libs/1_70_0/doc/html/boost_asio/reference/posix__basic_stream_descriptor/get_executor.html), so it can be used safely.

thed636 commented on tests/test_asio.h:
```diff
@@ -69,19 +72,27 @@ struct execution_context : asio::execution_context {
 
+        void assert_has_impl() const {
+            if (!impl_)
+                throw std::logic_error("ozo::testing::execution_context::executor_type::assert_impl() no executor mock");
```
Could you please place curly braces for if statement, like in project. The rationale for this are:
- changes inside `if ()` statement scope will not affect statement itself, so the history of the project will be more clear,
- sometimes weird  things like `if();` and similar are happened, so the braces help to avoid such thing.

thed636 reviewed (changes requested):
Thanks for the PR!

To make your first PR legally acceptable it is needed to add you into the [AUTHORS](https://github.com/yandex/ozo/blob/master/AUTHORS). The string with your name and e-mail would mean that you're agree with text on first two strings. It is necessary since the project is in the Yandex's  organisation.

JonasProgrammer committed:
Boost 1.70 compatibility
- get_io_context: use get_executor().context() to retrieve underlying context
- fix test_asio.hpp to always have an executor present (required for get_io_context()
- use 1.70 compatible version of resource_pool
- add branch to .gitmodules for simpler updates

Commit:
Merge branch 'master' into feature/boost_1_70_support
